spinoso-time/src/time/tzrs/mod.rs

```diff
@@ -0,0 +1,259 @@
+            to_a.month,
+            to_a.day,
+            to_a.hour,
+            to_a.min,
+            to_a.sec,
+            0,
+            Offset::from(to_a.zone),
+        )
+    }
+}
+
+// Core
+impl Time {
+    /// Returns the number of seconds as a signed integer since the Epoch.
+    ///
+    /// This function can be used to implement the ruby methods [`Time#to_i`] and [`Time#tv_sec`]
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use spinoso_time::tzrs::Time;
+    /// let t = Time::utc(1970, 1, 1, 0, 1, 0, 0);
+    /// assert_eq!(t.to_int(), 60)
+    /// ```
+    ///
+    /// [`Time#to_i`]: https://ruby-doc.org/core-2.6.3/Time.html#method-i-to_i
+    /// [`Time#tv_sec`]: https://ruby-doc.org/core-2.6.3/Time.html#method-i-tv_sec
+    #[inline]
+    #[must_use]
+    pub fn to_int(&self) -> i64 {
+        self.inner.unix_time()
+    }
+
+    /// Returns the number of seconds since the Epoch with fractional nanos included at IEEE
+    /// 754-2008 accuracy.
+    ///
+    /// This function can be used to implement the ruby method [`Time#to_f`]
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use spinoso_time::tzrs::Time;
+    /// let now = Time::utc(1970, 1, 1, 0, 1, 0, 1000);
+    /// assert_eq!(now.to_float(), 60.000001)
+    /// ```
+    ///
+    /// [`Time#to_f`]: https://ruby-doc.org/core-2.6.3/Time.html#method-i-to_f
+    #[inline]
+    #[must_use]
+    pub fn to_float(&self) -> f64 {
+        // A f64 mantissa is only 52 bits wide, so putting 64 bits in there will result in a
+        // rounding issues, however this is expected in the Ruby spec.
+        #[allow(clippy::cast_precision_loss)]
+        let sec = self.to_int() as f64;
+        let nanos_fractional = f64::from(self.inner.nanoseconds()) / f64::from(NANOS_IN_SECOND);
+        sec + nanos_fractional
+    }
+
+    /// Returns the numerator and denominator for the number of nano seconds of the Time struct
+    /// unsimplified.
+    ///
+    /// This can be used directly to implement [`Time#subsec`].
+    ///
+    /// This function can be used in combination with [`to_int`] to implement [`Time#to_r`].
+    ///
+    /// #Examples
+    ///
+    /// ```
+    /// use spinoso_time::tzrs::Time;
+    /// let t = Time::utc(1970, 1, 1, 0, 0, 1, 1000);
+    /// assert_eq!(t.subsec_fractional(), (1000, 1000000000));
+    /// ```
+    ///
+    /// [`Time#subsec`]: https://ruby-doc.org/core-2.6.3/Time.html#method-i-subsec
+    /// [`to_int`]: struct.Time.html#method.to_int
+    /// [`Time#to_r`]: https://ruby-doc.org/core-2.6.3/Time.html#method-i-to_r
+    #[inline]
+    #[must_use]
+    pub fn subsec_fractional(&self) -> (u32, u32) {
+        (self.inner.nanoseconds(), NANOS_IN_SECOND)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    fn time_with_fixed_offset(offset: i32) -> Time {
+        let offset = Offset::fixed(offset);
+        Time::with_timespec_and_offset(0, 0, offset)
+    }
+
+    #[test]
+    fn time_zone_fixed_offset() {
```

Comment:
thank you so much for getting offsets working!

Reply:
You would not believe how much of a headache it was. This was iteration like 4 or 5 :grimacing:. I'm mostly certain this will be enough for a full `Time#at` impl.